doc/architecture.rst

```diff
@@ -369,7 +369,39 @@ is renamed or moved.
+
+The first time E2EE is enabled on a folder in any of the clients, the user is prompted with a private key consisting of 12 security words. The user is strongly recommended to record these somewhere secure as the complete loss of this private key means there is no way to access their data anymore. The key is also securily stored in the device's key storage and can be shown on demand. Making the folder available on a second device requires entering the private key. Future versionf of the Nextcloud clients will be able to display a QR code to simplify adding devices. Sharing with other users will not require any special keys or passwords.
+
+Encrypting files locally means the server has no access to them. This brings with it a number of limitations:
+
+* E2EE files can not be accessed or previewed through the web interface
+* E2EE files can not be edited with Online Office solutions
+* E2EE files can not be shared with a public link
+* E2EE files can not be searched, tagged, commented on and have no versioning or trash bin
+* E2EE files can not be accessed in other Nextcloud Apps. This means they have no chat sidebar, can not be attached to emails or deck cards, shared in Talk rooms and so on
+* E2EE results in slower syncing of file and works poorly or not at all with large files and large quantities of files
+
+These limitations are fundamental to how securely implemented end-to-end encryption works. Offering browser access to end-to-end encrypted files would essentially negate any of the benefits of end-to-end encryption. Serving a file in the browser would mean the server needs to be able to read the files, and thusly, the administrator or a maliscious attacker who gained access to the server, can, too. Decrypting the file in the browser does not solve this security risk in the least, as the javascript code that would be needed to decrypt the file comes FROM the server, and of course a compromised server would simply send modified javascript code which sends a copy of the encryption keys to the attacker without anybody noticing. See for more details our blog about the `threat model for the encryption solutions in Nextcloud`_ and our `webpage about End-to-end Encryption`_.
+
+The E2EE design of Nextcloud allows for sharing on a per-folder level to individual users (not groups), but, as of early 2021, this feature is still on the roadmap for implementation in the clients.
+
+Due to all these limitations that come with end-to-end encryption, it is only recommended for a small sub-set of files, in just a small number of folders. Encryptiong your entire sync folder is likely to result in poor performance and sync errors and if you do not trust your server at all, Nextcloud is perhaps not the right solution for your use case.
```

Comment:
@jospoortvliet 

Nice! I would also suggest adding something like this:

In case you still want to encrypt a large file or folder, it is recommended to compress it to a password-protected archive, and then, create a text file with a password to that archive. Then, you'd need to encrypt that password-containing file with E2EE. With this approach, you can be sure your archive that contains sensitive data is still encrypted by means of archive password protection, but, the end-to-end encryption is only applied to a small text file that can be used on your local device to extract the password-protected archive.

Reply:
> @jospoortvliet
> 
> Nice! I would also suggest adding something like this:
> 
> In case you still want to encrypt a large file or folder, it is recommended to compress it to a password-protected archive, and then, create a text file with a password to that archive. Then, you'd need to encrypt that password-containing file with E2EE. With this approach, you can be sure your archive that contains sensitive data is still encrypted by means of archive password protection, but, the end-to-end encryption is only applied to a small text file that can be used on your local device to extract the password-protected archive.

Pfff, at that point, I'd say "use a password manager" or an encrypted partition or one of a dozen other solutions. I think that saying that Nextcloud is perhaps not the right solution should get the message across, giving alternative solutions isn't really worth the space in our documentation. Hmmm. I could just say 'use password protected archives or other solutions", let me see if I can reword it a bit. Tnx!